libazureinit/src/media.rs

```diff
@@ -66,27 +68,94 @@ fn default_preprov_type() -> String {
     "None".to_owned()
 }
 
-pub fn make_temp_directory() -> Result<(), Box<dyn std::error::Error>> {
-    let file_path = "/run/azure-init/tmp/";
+pub const PATH_MOUNT_DEVICE: &str = "/dev/sr0";
+pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 
-    create_dir_all(file_path)?;
+// Some zero-sized structs that just provide states for our state machine
+pub struct Mounted;
+pub struct Unmounted;
 
-    let metadata = fs::metadata(file_path)?;
-    let permissions = metadata.permissions();
-    let mut new_permissions = permissions.clone();
-    new_permissions.set_mode(0o700);
-    fs::set_permissions(file_path, new_permissions)?;
+pub struct Media<State = Unmounted> {
+    device_path: PathBuf,
+    mount_path: PathBuf,
+    state: std::marker::PhantomData<State>,
+}
+
+impl Media<Unmounted> {
+    pub fn new(device_path: PathBuf, mount_path: PathBuf) -> Media<Unmounted> {
+        Media {
+            device_path,
+            mount_path,
+            state: std::marker::PhantomData,
+        }
+    }
+
+    // The only thing you can do with the Media struct is call mount which, if it succeeds,
+    // moves it into the Mounted state. From there the only thing you can do with it is
+    // call unmount(), but you could make other functionality available by adding to the
+    // impl Media<Mounted> block.
```

Comment:
This comment is outdated now that you added `read_ovf_env_to_string()`. I think you can just delete it and we can have a task to come through and document the library's API in a follow-up PR.

Reply:
Deleted